src/nile/core/deploy.py

```diff
@@ -17,18 +26,27 @@ def deploy(
 ):
     """Deploy StarkNet smart contracts."""
     logging.info(f"🚀 Deploying {contract_name}")
+
     base_path = (
         overriding_path if overriding_path else (BUILD_DIRECTORY, ABIS_DIRECTORY)
     )
     register_abi = abi if abi is not None else f"{base_path[1]}/{contract_name}.json"
 
-    output = run_command(
-        operation="deploy",
-        network=network,
-        contract_name=contract_name,
-        overriding_path=overriding_path,
-        inputs=arguments,
-        mainnet_token=mainnet_token,
+    contract = f"{base_path[0]}/{contract_name}.json"
+    command_args = ["--contract", contract]
+
+    if arguments:
+        command_args.append("--inputs")
+        command_args.extend(prepare_params(arguments))
+
+    if mainnet_token:
+        command_args.append("--token")
+        command_args.extend(mainnet_token)
+
+    args = set_args(network)
+
+    output = await capture_stdout(
+        starknet_cli.deploy(args=args, command_args=command_args)
```

Comment:
can't we abstract this with something like:
```python
def call_cli(cmd_name, args, command_args):
    cmd = getattr(starknet_cli, cmd_name)
    return capture_stdout(
        cmd(args=args, command_args=command_args)
    )

Reply:
Good call, that should work!

Reply:
does it make sense to abstract `starknet_cli` into its own module exposing the commands?

Reply:
The only potential benefit I see right now is removing reliance on common.py (which might be worth doing).